src/app/WriteHandler.cpp

```diff
@@ -0,0 +1,315 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines write handler for a CHIP Interaction Data model
+ *
+ */
+
+#include <app/InteractionModelEngine.h>
+#include <app/MessageDef/EventPath.h>
+#include <app/WriteHandler.h>
+#include <app/reporting/Engine.h>
+
+namespace chip {
+namespace app {
+CHIP_ERROR WriteHandler::Init(InteractionModelDelegate * apDelegate)
+{
+    VerifyOrReturnError(apDelegate != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mpExchangeCtx == nullptr, CHIP_ERROR_INCORRECT_STATE);
+    mpExchangeCtx = nullptr;
+    mpDelegate    = apDelegate;
+
+    System::PacketBufferHandle packet = System::PacketBufferHandle::New(chip::app::kMaxSecureSduLengthBytes);
+    VerifyOrReturnError(!packet.IsNull(), CHIP_ERROR_NO_MEMORY);
+
+    mMessageWriter.Init(std::move(packet));
+    ReturnErrorOnFailure(mWriteResponseBuilder.Init(&mMessageWriter));
+
+    AttributeStatusList::Builder attributeStatusListBuilder = mWriteResponseBuilder.CreateAttributeStatusListBuilder();
+    ReturnErrorOnFailure(attributeStatusListBuilder.GetError());
+
+    MoveToState(State::Initialized);
+
+    return CHIP_NO_ERROR;
+}
+
+void WriteHandler::Shutdown()
+{
+    VerifyOrReturn(mState != State::Uninitialized);
+    mMessageWriter.Reset();
+    ClearExistingExchangeContext();
```

Comment:
similar to other comments: either make this void (cannot fail) or check and use error code return.

Reply:
Done, thanks